src/Kernels/Distance/Cosine.php

```diff
@@ -52,9 +52,15 @@ public function compute(array $a, array $b) : float
             $valueB = $b[$i];
 
             $sigma += $valueA * $valueB;
+        }
+
+        if (abs($sigma) > EPSILON) {
```

Comment:
I think we'll be ok with just checking for 0 instead of +/- EPSILON since even after TF-IDF words that do not appear in a document are still 0.

<img width="184" alt="Annotation 2020-08-26 194345" src="https://user-images.githubusercontent.com/18690561/91371559-efc67600-e7d6-11ea-8b17-cbc0fa795dc3.png">